presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -55,103 +71,198 @@ public CatalogServer(MetadataManager metadataProvider, SessionPropertyManager se
         this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
         this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
         this.objectMapper = requireNonNull(objectMapper, "handleResolver is null");
+
+        OptionalLong cacheExpiresAfterWriteMillis = OptionalLong.of(600000);
```

Comment:
Let's create constants for these magic numbers. Also maybe add a TODO to make them configurable.